### PR TITLE
Switch MetadataApi and it's usage to new sync CallApi

### PR DESCRIPTION
**Pull request by LiubovDidkivska**
Relates-To: OLPEDGE-1190
Signed-off-by: Liubov Didkivska <ext-liubov.didkivska@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp [hunk 1]
```diff
@@ -55,110 +55,99 @@ namespace dataservice {
...
-      layerVersionsCallback(
-          olp::parser::parse<model::LayerVersions>(response.response));
-    }
-  };
-
-  return client.CallApi(metadataUri, "GET", queryParams, headerParams,
-                        formParams, nullptr, "", callback);
+  auto apiResponse =
+          client.CallApi(metadataUri, "GET", queryParams, headerParams,
+  {}, nullptr, "", context);
```
Comment (diachenko-mischa) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp]: clang-format

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp [hunk 2]
```diff
@@ -55,110 +55,99 @@ namespace dataservice {
...
 
-CancellationToken MetadataApi::GetLayerVersions(
+MetadataApi::LayerVersionsResponse MetadataApi::GetLayerVersions(
     const OlpClient& client, int64_t version,
     boost::optional<std::string> billingTag,
-    const LayerVersionsCallback& layerVersionsCallback) {
+    client::CancellationContext context) {
   std::multimap<std::string, std::string> headerParams;
-  headerParams.insert(std::make_pair("Accept", "application/json"));
+  headerParams.emplace("Accept", "application/json");
```
Comment (diachenko-mischa) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp]: Update code to google code style rules

```suggestion
  header_params.emplace("Accept", "application/json");
```

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp [hunk 3]
```diff
@@ -55,110 +55,99 @@ namespace dataservice {
...
-  return client.CallApi(metadataUri, "GET", queryParams, headerParams,
-                        formParams, nullptr, "", callback);
+  auto apiResponse =
+          client.CallApi(metadataUri, "GET", queryParams, headerParams,
+  {}, nullptr, "", context);
+
+  auto strResponse = apiResponse.response.str();
+
+  if (apiResponse.status != http::HttpStatusCode::OK) {
+       return ApiError(apiResponse.status, strResponse);
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp]: we should move strResponse

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp [hunk 4]
```diff
@@ -55,110 +55,99 @@ namespace dataservice {
...
+  }
+
+  return LayerVersionsResponse(olp::parser::parse<model::LayerVersions>(apiResponse.response));
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp]: same

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.h [hunk 5]
```diff
@@ -54,26 +55,24 @@ class MetadataApi {
   using LayerVersionsCallback = std::function<void(LayerVersionsResponse)>;
 
   /**
-   * @brief Call to asynchronously retrieve the latest metadata version for each
-   * layer of a specified catalog metadata version.
+   * @brief Retrieve the latest metadata version for each layer of a specified
```
Comment (diachenko-mischa) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.h]:
```suggestion
   * @brief Retrieves the latest metadata version for each layer of a specified
```

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.h [hunk 6]
```diff
@@ -54,26 +55,24 @@ class MetadataApi {
...
+  static LayerVersionsResponse GetLayerVersions(
       const client::OlpClient& client, int64_t version,
       boost::optional<std::string> billingTag,
-      const LayerVersionsCallback& callback);
+      client::CancellationContext context);
 
   /**
-   * @brief Call to asynchronously retrieve metadata for all partitions in a
-   * specified layer.
+   * @brief Retrieve metadata for all partitions in a specified layer.
```
Comment (diachenko-mischa) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.h]:
```suggestion
   * @brief Retrieves metadata for all partitions in a specified layer.
```

File: olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.cpp [hunk 7]
```diff
@@ -166,44 +164,8 @@ CatalogVersionResponse CatalogRepository::GetLatestVersion(
...
-    cancellation_context.CancelOperation();
-    OLP_SDK_LOG_INFO_F(kLogTag, "timeout");
-    return ApiError(ErrorCode::RequestTimeout, "Network request timed out.");
-  }
-  interest_flag->store(false);
-
-  if (cancellation_context.IsCancelled()) {
-    return ApiError(ErrorCode::Cancelled, "Operation cancelled.");
-  }
+  MetadataApi::CatalogVersionResponse version_response = MetadataApi::GetLatestCatalogVersion(
```
Comment (diachenko-mischa) [inline olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.cpp]:
```suggestion
  auto version_response = MetadataApi::GetLatestCatalogVersion(
```

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp [hunk 8]
```diff
@@ -55,110 +55,99 @@ namespace dataservice {
...
-    }
-  };
-
-  return client.CallApi(metadataUri, "GET", queryParams, headerParams,
-                        formParams, nullptr, "", callback);
+  auto apiResponse =
+          client.CallApi(metadataUri, "GET", queryParams, headerParams,
+  {}, nullptr, "", context);
+
+  auto strResponse = apiResponse.response.str();
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp]: should move under the if statement

File: olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp [hunk 9]
```diff
@@ -146,48 +146,9 @@ PartitionsResponse PartitionsRepository::GetPartitions(
...
-
-  flag->store(false);
-
-  if (cancellation_context.IsCancelled()) {
-    // We can't use api response here because it could potentially be
-    // uninitialized.
-    return client::ApiError(client::ErrorCode::Cancelled,
-                            "Operation cancelled.");
-  }
+  PartitionsResponse metadata_response = MetadataApi::GetPartitions(
```
Comment (diachenko-mischa) [inline olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp]:
```suggestion
  auto metadata_response = MetadataApi::GetPartitions(
```

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp [hunk 10]
```diff
@@ -55,110 +55,99 @@ namespace dataservice {
...
+
+  if (apiResponse.status != http::HttpStatusCode::OK) {
+       return ApiError(apiResponse.status, strResponse);
+  }
+
+  return PartitionsResponse(olp::parser::parse<model::Partitions>(apiResponse.response));
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp]: move

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp [hunk 11]
```diff
@@ -55,110 +55,99 @@ namespace dataservice {
...
-    }
-  };
-
-  return client.CallApi(metadataUri, "GET", queryParams, headerParams,
-                        formParams, nullptr, "", callback);
+  auto apiResponse =
+          client.CallApi(metadataUri, "GET", queryParams, headerParams,
+  {}, nullptr, "", context);
+
+  auto strResponse = apiResponse.response.str();
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp]: should move under the if statement

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp [hunk 12]
```diff
@@ -55,110 +55,99 @@ namespace dataservice {
...
+
+  if (apiResponse.status != http::HttpStatusCode::OK) {
+       return ApiError(apiResponse.status, strResponse);
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp]: std::move(strResponse)

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp (hunk 10 above)
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp]: std::move(apiResponse.response)

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp [hunk 13]
```diff
@@ -55,110 +55,99 @@ namespace dataservice {
...
+  }
+  return CatalogVersionResponse(olp::parser::parse<model::VersionResponse>(apiResponse.response));
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.cpp]: same as above

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.h [hunk 14]
```diff
@@ -54,26 +55,24 @@ class MetadataApi {
   using LayerVersionsCallback = std::function<void(LayerVersionsResponse)>;
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.h]: We can now remove callback types

File: olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.cpp (hunk 7 above)
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.cpp]: can be just auto

File: olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp (hunk 9 above)
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp]: same

File: olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp [hunk 15]
```diff
@@ -146,48 +146,9 @@ PartitionsResponse PartitionsRepository::GetPartitions(
...
+              query_api.GetResult(), layer, request.GetVersion(),
+              boost::none, boost::none, request.GetBillingTag(),cancellation_context);
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp]: please use clang-format

**Pull request merged**